### DISCUSSION TITLE
How to use global variables in a input port handle?

### DISCUSSION THREAD

**Discussion by zyrfyxx** [nasa/fprime#1960] [Q&A] [answered]
If I want to use and change a global variable in different component, how to create the global var?
If I create a struct type, then use it as a global and get&set its value in different component, where I put this type?
Afternow, I can only pass a variable to the input handle through the port parameter, can't use a global variable.

**Comment by timcanham** [comment, marked as answer]
One of the architectural principals of F Prime is that you want the behaviors in components decoupled, where interactions are only via clearly-defined port calls. Sharing a global variable to avoid port calls violates that principle. If you need to share a value, you can have one component as the owner, and have a port call to retrieve that value. If it is a scalar, you can also use `Svc/PolyDb` to share values between components.
